src/twisted/mail/test/test_mail.py

```diff
@@ -247,25 +248,80 @@ def _cbFinalName(self, result):
         self.assertTrue(self.f.closed)
         self.assertFalse(os.path.exists(self.name))
 
-    @skipIf(sys.version_info >= (3,), "not ported to Python 3")
     def testContents(self):
-        contents = "first line\nsecond line\nthird line\n"
+        contents = b"first line\nsecond line\nthird line\n"
         for line in contents.splitlines():
             self.fp.lineReceived(line)
         self.fp.eomReceived()
-        with open(self.final) as f:
+        with open(self.final, "rb") as f:
             self.assertEqual(f.read(), contents)
 
-    @skipIf(sys.version_info >= (3,), "not ported to Python 3")
     def testInterrupted(self):
-        contents = "first line\nsecond line\n"
+        contents = b"first line\nsecond line\n"
         for line in contents.splitlines():
             self.fp.lineReceived(line)
         self.fp.connectionLost()
         self.assertFalse(os.path.exists(self.name))
         self.assertFalse(os.path.exists(self.final))
 
 
+@skipIf(platformType != "posix", "twisted.mail only works on posix")
+class MaildirMessageTests(TestCase):
+    def setUp(self):
+        self.name = "maildirMessage.testFile"
+        self.final = "final.maildirMessage.testFile"
+        self.address = b"user@example.com"
+        self.f = open(self.name, "wb")
```

Comment:
Rather than closing the file in `tearDown`, use [`SynchronousTestCase.addCleanup`](https://twistedmatrix.com/documents/current/api/twisted.trial.unittest.SynchronousTestCase.html#addCleanup). Then you don't need to maintain parallel structure between `setUp` and `tearDown`.

```suggestion
        self.f = open(self.name, "wb")
        self.addCleanup(self.f.close)
```